src/NotificationTemplate.php

```diff
@@ -423,6 +423,13 @@ public static function process($string, $data)
             }
         }
 
+        // Convert domain relative links to absolute links
+        $cleandata = preg_replace(
+            '/((?:href)=[\'"])(\/[^\/].*)([\'"])/',
+            '$1' . $CFG_GLPI['url_base'] . '$2$3',
+            $cleandata
+        );
+
```

Comment:
It should be moved just before the return part. Indeed, `$string = strtr($string, $cleandata);` may add some relative URLs to the content, and they must be replaced too.

Reply:
Ignore previous comment, I misread the code.

To optimize the replacement, maybe it could still be done only on the final `$string`. Indeed, here you are doing it on many data that will not be necessarly be present in the final string.